packages/drivers/odsp-driver/src/odspDocumentDeltaConnection.ts

```diff
@@ -366,10 +366,16 @@ export class OdspDocumentDeltaConnection extends DocumentDeltaConnection {
 		// Note: we suspect the incoming error object is either:
 		// - a socketError: add it to the OdspError object for driver to be able to parse it and reason over it.
 		// - anything else: let base class handle it
-		// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-		return canRetry && Number.isInteger(error?.code) && typeof error?.message === "string"
-			? errorObjectFromSocketError(error as IOdspSocketError, handler)
-			: super.createErrorObject(handler, error, canRetry);
+		const errorObject =
+			// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+			canRetry && Number.isInteger(error?.code) && typeof error?.message === "string"
+				? errorObjectFromSocketError(error as IOdspSocketError, handler)
+				: super.createErrorObject(handler, error, canRetry);
+
+		// We use errorFrom param to clear the joinSession cache if the error happens in connect_document flow.
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-member-access
+		(errorObject as any).errorFrom = handler;
```

Comment:
This time I have made the change in the overridden function of the `odspDocumentDeltaConnection.ts` instead of the base class of `documentDeltaConnection`.

Reply:
How about adding this prop to the IDriverErrorBase or IAnyDriverError, so that we don't have to cast it to any?

Reply:
IDriverErrorBase / IAnyDriverError - seem to be a generic interfaces. I am hesitant in making any ODSP driver specific changes there. @jatgarg  - Do you agree?

Reply:
This is not ODSP specific because the base class from where this error comes is used by other drivers too like r11s driver. We can add more generalized/informative field rather than "errorFrom" and then add this on the socket error maybe?

Reply:
Fixed now, please re-review @jatgarg